src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -0,0 +1,16 @@
+package cromwell.engine.backend
+
+import cromwell.binding.values.WdlValue
+import cromwell.binding.{Call, WdlStandardLibraryFunctions, WorkflowDescriptor}
+
+import scala.util.Try
+
+trait BackendCall {
```

Comment:
I'll try to check out this branch over the weekend and see if what I was suggesting appears to work before getting too far down this rabbit hole, but it seems to be exactly the problem an f-bound is trying to solve.

The idea of the code above is that it's a recursive bound. It's saying that `BackendType` takes a type parameter that must be a sub-type of `BackendType` infused with that sub-type. If only github had that mind blown emoji. 

At that point you can use that sub-type (`A`) in your trait and while there are simpler ways to set this up the rest all lead to potential oddities and shenanigans. 

Here's a lightning talk on the topic from this fall, just 15 mins:
https://newcircle.com/s/post/1717/f_bounded_polymorphism_marconi_lanna_video
At first I loved the fact that the dude's slides were written in the REPL but it did get annoying after a while